activitysim/core/chunk.py

```diff
@@ -1260,6 +1267,14 @@ def adaptive_chunked_choosers(
     while offset < num_choosers:
         i += 1
 
+        if (
+            explicit_and_odd_num_choosers
+            & (i == estimated_number_of_chunks)
+            & (rows_per_chunk > 1)
+        ):
+            # last chunk may be smaller than chunk_size due to rounding error
```

Comment:
We don't need to update the rows_per_chunk here, as noted above we can overrun the end of the choosers and be fine.

Reply:
see above response.